MapboxCoreNavigation/sequence.swift

```diff
@@ -0,0 +1,11 @@
+import Foundation
+
+extension Sequence where Element: Hashable {
+    func compatibleFlatMap<ElementOfResult>(_ transform: (Element) throws -> ElementOfResult?) rethrows -> [ElementOfResult] {
+        #if swift(>=4.1)
```

Comment:
I'm also not crazy about using static version checks -- as an old ObjC'er, I prefer to check for API availability. Just my 2¢, but if we address my other comment we will be effectively doing that, so...